### ISSUE TITLE
✏️ [ FEAT REQ ] New "STOP" Notification Action

### ISSUE THREAD

**Issue by adamhotep** [octoshrimpy/quik#62]
I receive lots of spim ([messaging spam](https://en.wikipedia.org/wiki/Messaging_spam)) and would like to better dispatch it. In Notifications → Actions, there is an option for _Reply_ but that pops up a reply dialog. 

I propose an option called _STOP_ that will:
1. Reply with the word "STOP" in all-caps without the quotes
2. Mark the message as read
3. Suppress message notifications from that sender for 30 seconds (or 1 minute?)

I suppose this could be configurable. Some users might want to block the sender rather than temporarily muting them. Others may not want to mark the message as read.

**Comment by lawson85** [comment]
MacroDroid is free and can do this in the mean time

**Comment by octoshrimpy** [comment]
@adamhotep heads up, friend:
https://www.reddit.com/r/YouShouldKnow/comments/x2q37p/ysk_you_shouldnt_reply_stop_to_spam_text_messages/

You shouldn't be replying to those messages. they are sent en masse, and replying just tells them there's an interactive human on the other side. I've heard rumors of spammers using SMSes like that to filter their list for more vulnerable targets.

**Comment by octoshrimpy** [comment]
although in spirit it does tie into #161

**Comment by adamhotep** [comment]
@octoshrimpy I'm in the anti-spam industry, and while I focus on email, I have industry friends who deal with spim, aka messaging spam. The past wisdom of avoiding unsubscribe isn't as valid nowadays as it was 20 years ago. Since I'm knee-deep in email anti-spam, I have the tools and know-how to figure out which subscription mechanisms are worth using (lots of spam is sent as platform abuse, so the unsubscribe action actually helps the platform provider identify and discipline their problematic customers). 

On the SMS side, there isn't enough information to do this, so it does indeed make more sense to be careful. I don't mind the risk because (1) it's trivial to confirm that a phone number is an active cell phone and (2) I also report all of my spam to 7726 (which I've been meaning to write up as a feature request for Quik).

The reason I created this request was that a peer at AT&T confirmed my suspicion about how they handle reports of political spam, which is to say that they do not. The *only* recourse for these is to respond with "STOP". I can at least anecdotally confirm that responding "STOP" does help; I used to get repeats from senders (on different phone numbers, so blocking does not work) based on their link domains.

For example, dem-strategy.org sent me political SMS spams on Apr 15, Apr 18, Apr 22, and Apr 24, all from different numbers. I reported each of them to 7726 and otherwise ignored the first three. I actually responded "Stop" to the fourth, and while they never confirmed that, I never heard from them again.
